src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisDTOService.java

```diff
@@ -0,0 +1,160 @@
+                .orElse(Map.of());
+        return new PyrisSubmissionDTO(submission.getId(), toInstant(submission.getSubmissionDate()), studentRepositoryContents, submission.getParticipation().isPracticeMode(),
+                submission.isBuildFailed(), buildLogEntries, getLatestResult(submission));
+    }
+
+    /**
+     * Convert a list of IrisMessage to a list of PyrisMessageDTO
+     *
+     * @param messages the messages
+     * @return the PyrisMessageDTOs
+     */
+    public List<PyrisMessageDTO> toPyrisDTO(List<IrisMessage> messages) {
+        return messages.stream().map(message -> {
+            var content = message.getContent().stream().map(messageContent -> {
+                PyrisMessageContentDTO result = null;
+                if (messageContent.getClass().equals(IrisTextMessageContent.class)) {
+                    result = new PyrisTextMessageContentDTO(messageContent.getContentAsString());
+                }
+                else if (messageContent.getClass().equals(IrisJsonMessageContent.class)) {
+                    result = new PyrisJsonMessageContentDTO(messageContent.getContentAsString());
+                }
+                return result;
+            }).filter(Objects::nonNull).toList();
+            return new PyrisMessageDTO(toInstant(message.getSentAt()), message.getSender(), content);
+        }).toList();
+    }
+
+    /**
+     * Null safe conversion of ZonedDateTime to Instant
+     *
+     * @param zonedDateTime the ZonedDateTime to convert
+     * @return the Instant or null if the input was null
+     */
+    private Instant toInstant(ZonedDateTime zonedDateTime) {
+        if (zonedDateTime == null) {
+            return null;
+        }
+        return zonedDateTime.toInstant();
+    }
+
+    /**
+     * Helper method to convert the latest result of a submission to a PyrisResultDTO
+     *
+     * @param submission the submission
+     * @return the PyrisResultDTO or null if the submission has no result
+     */
+    private PyrisResultDTO getLatestResult(ProgrammingSubmission submission) {
+        var latestResult = submission.getLatestResult();
+        if (latestResult == null) {
+            return null;
+        }
+        var feedbacks = latestResult.getFeedbacks().stream().map(feedback -> {
+            var text = feedback.getDetailText();
+            if (feedback.getHasLongFeedbackText()) {
+                text = feedback.getLongFeedback().get().getText();
+            }
+            var testCaseName = feedback.getTestCase() == null ? feedback.getText() : feedback.getTestCase().getTestName();
+            return new PyrisFeedbackDTO(text, testCaseName, Objects.requireNonNullElse(feedback.getCredits(), 0D));
+        }).toList();
+
+        return new PyrisResultDTO(toInstant(latestResult.getCompletionDate()), latestResult.isSuccessful(), feedbacks);
+    }
+
+    /**
+     * Helper method to get & checkout the repository for a participation
```

Comment:
Documentation can also be improved here, in particular why this would be necessary (i.e. the rationale)